anndata/_core/merge.py

```diff
@@ -1239,7 +1245,9 @@ def concat(
         [pd.Series(dim_indices(a, axis=axis)) for a in adatas], ignore_index=True
     )
     if index_unique is not None:
-        concat_indices = concat_indices.str.cat(label_col.map(str), sep=index_unique)
+        concat_indices = concat_indices.str.cat(
+            label_col.map(str, na_action="ignore"), sep=index_unique
+        )
```

Comment:
```suggestion
        if parse_version(pd.__version__) >= parse_version("2.0"):
            # Argument added in pandas 2.0
            concat_indices = concat_indices.str.cat(
                label_col.map(str, na_action="ignore"), sep=index_unique
            )
        else:
            concat_indices = concat_indices.str.cat(
                label_col.map(str), sep=index_unique
            )
```

It looks like `na_value` wasn't defined pre 2.0 and unfortunately I don't think we can just bump our minimum pandas version, since it looks like uptake of pandas 2.0 is [unfortunately slow](https://www.pepy.tech/projects/pandas?versions=2.*&versions=1.*&versions=*)

You could probably cherry pick this commit: https://github.com/scverse/anndata/pull/1189/commits/d7d0fa7eca2694325af1ae609116a24e069a0230

Reply:
merged your main changes